server/src/configuration_set.rs

```diff
@@ -49,12 +66,22 @@ impl ConfigurationSet {
         slice_config.set_built_in_slice_path(include_built_in.then(|| built_in_path.to_owned()));
         slice_config.set_search_paths(paths);
 
-        let options = slice_config.as_slice_options();
-        let compilation_state = slicec::compile_from_options(options, |_| {}, |_| {});
-        Self {
-            slice_config,
-            compilation_state,
-        }
+        let compilation_data = CompilationData::default();
+        Self { slice_config, compilation_data }
+    }
+
+    pub fn trigger_compilation(&mut self) -> Vec<Diagnostic> {
+        // Perform the compilation.
+        let slice_options = self.slice_config.as_slice_options();
+        let compilation_state = slicec::compile_from_options(slice_options, |_| {}, |_| {});
+        let CompilationState { ast, diagnostics, files } = compilation_state;
+
+        // Process the diagnostics (filter out allowed lints, and update diagnostic levels as necessary).
+        let updated_diagnostics = diagnostics.into_updated(&ast, &files, slice_options);
```

Comment:
No additional information is provided in `into_updated(&ast, &files, slice_options)` that wasn't already provided in the original call to `slicec::compile_from_options(slice_options, |_| {}, |_| {})`. Since `slice_options` also should not have mutated between line 75-80 is this call to `into_updated` necessary?